src/api/app/assets/javascripts/webui/application.js

```diff
@@ -246,3 +246,36 @@ function select_from_autocomplete(toselect) {
         if ($(this).text() == toselect) { $(this).trigger('mouseenter').click(); }
     });
 }
+
+// convert timestamps to real dates
+$(function() {
```

Comment:
This is A LOT js code for converting time which I think is not necessary.
What about this solution:
1) Your create two html elements, one with fuzzy time and one with full time stamp. You assign different classes to and hide the full time element by default.
```
<span class='fuzzy-time'> <%= fuzzy_time(Time.at(commit['time'].to_i)) %> (revision <%= commit['rev'] %>) </span>
<span class='full-time hidden'> <%= commit['time'].to_i %> (revision <%= commit['rev'] %>) </span>
```
2) In your javascript you toggle when clicking on the link
```
$( ".fuzzy-time" ).toggle();
$( ".normal-time" ).toggle();
```
No loops, no storing of data in data-elements, less code :tada: 
What do you think?